src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -19644,32 +19647,228 @@ static void F(C<object>? x, C<object?> y, bool c)
+                );
+        }
+
+        [Fact]
+        public void SuppressNullableWarning_NullCoalescing()
+        {
+            var source =
+@"
+class C<T>
+{
+    static void F(C<object>? x, C<object?> y)
+    {
+        var t1 = x ?? y;
+        var t2 = y ?? x;
+        var t3 = x! ?? y;
+        var t4 = y! ?? x;
+        var t5 = x ?? y!;
+        var t6 = y ?? x!;
+        var t7 = x! ?? y!;
+        var t8 = y! ?? x!;
+    }
+}";
+            var comp = CreateCompilation(
+                source,
+                parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (6,23): warning CS8619: Nullability of reference types in value of type 'C<object?>' doesn't match target type 'C<object>'.
+                //         var t1 = x ?? y;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "y").WithArguments("C<object?>", "C<object>").WithLocation(6, 23),
+                // (7,18): hidden CS8607: Expression is probably never null.
+                //         var t2 = y ?? x;
+                Diagnostic(ErrorCode.HDN_ExpressionIsProbablyNeverNull, "y").WithLocation(7, 18),
+                // (7,23): warning CS8619: Nullability of reference types in value of type 'C<object>' doesn't match target type 'C<object?>'.
+                //         var t2 = y ?? x;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("C<object>", "C<object?>").WithLocation(7, 23),
+                // (8,18): hidden CS8607: Expression is probably never null.
+                //         var t3 = x! ?? y;
+                Diagnostic(ErrorCode.HDN_ExpressionIsProbablyNeverNull, "x!").WithLocation(8, 18),
+                // (9,18): hidden CS8607: Expression is probably never null.
+                //         var t4 = y! ?? x;
+                Diagnostic(ErrorCode.HDN_ExpressionIsProbablyNeverNull, "y!").WithLocation(9, 18),
+                // (11,18): hidden CS8607: Expression is probably never null.
+                //         var t6 = y ?? x!;
+                Diagnostic(ErrorCode.HDN_ExpressionIsProbablyNeverNull, "y").WithLocation(11, 18),
+                // (12,18): hidden CS8607: Expression is probably never null.
+                //         var t7 = x! ?? y!;
+                Diagnostic(ErrorCode.HDN_ExpressionIsProbablyNeverNull, "x!").WithLocation(12, 18),
+                // (13,18): hidden CS8607: Expression is probably never null.
+                //         var t8 = y! ?? x!;
+                Diagnostic(ErrorCode.HDN_ExpressionIsProbablyNeverNull, "y!").WithLocation(13, 18)
+                );
+        }
+
+        [Fact]
+        public void SuppressNullableWarning_ArrayInitializer()
+        {
+            var source =
+@"
+class C<T>
+{
+    static void F(C<object>? x, C<object?> y)
+    {
+        var a1 = new[] { x, y }; // 1
+        var a2 = new[] { x!, y };
```

Comment:
Shouldn't this result in a nested nullability mismatch warning since the items are `C<object!>` and `C<object?>!`?

Reply:
I missed this comment. I'll investigate.

---
In reply to: [192416347](https://github.com/dotnet/roslyn/pull/27317#discussion_r192416347) [](ancestors = 192416347)

Reply:
This is a known issue (PROTOTYPE marker in  VisitArrayInitializer). I'll add a comment here and look at it in a follow-up.

---
In reply to: [193619244](https://github.com/dotnet/roslyn/pull/27317#discussion_r193619244) [](ancestors = 193619244,192416347)